File: app/src/main/java/com/salman/bitclock/data/models/Timer.java
package com.salman.bitclock.data.models;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "timers")
public class Timer {
    @PrimaryKey(autoGenerate = true)
    public int id;

    public String name;
    public long initialDurationMs;
    public long remainingMs;
    public int status; // e.g., 0 = stopped, 1 = running
}

File: app/src/main/kotlin/com/salman/bitclock/data/models/Timer.kt
package com.salman.bitclock.data.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "timers")
data class Timer(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Int = 0,

    @ColumnInfo(name = "name")
    val name: String,

    @ColumnInfo(name = "initial_duration")
    val initialDurationMs: Long,

    @ColumnInfo(name = "remaining_time")
    val remainingMs: Long,

    @ColumnInfo(name = "status")
    val status: Int = 0 // 0 = stopped, 1 = running
)
